### PR TITLE
[Feature] Add adjust sharpness

### PR DESCRIPTION
**Pull request by LXXXXR**
Hi, 
In this pull request, `adjust_sharpness` is added for data augmentation. Pls kindly take a look. Thx.

### CODE REVIEW

File: mmcv/image/photometric.py [hunk 1]
```diff
@@ -227,6 +227,33 @@ def adjust_contrast(img, factor=1.):
...
+    ``output = img * factor + degenerated * (1 - factor)``
+
+    Args:
+        img (ndarray): Image to be sharpened. BGR order.
+        factor (float): Same as :func:`mmcv.adjust_brightness`.
+
+    Returns:
+        ndarray: The sharpened image.
+    """
+    kernel = np.array([[1., 1., 1.], [1., 5., 1.], [1., 1., 1.]]) / 13
```
Comment (MeowZheng) [inline mmcv/image/photometric.py:248]: Could you please give an explanation about kernel value?

**Review by MeowZheng — APPROVED** [review]

File: mmcv/image/photometric.py [hunk 2]
```diff
@@ -227,6 +227,34 @@ def adjust_contrast(img, factor=1.):
...
+def adjust_sharpness(img, factor=1.):
+    """Adjust image sharpness.
+
+    This function controls the sharpness of an image. An
+    enhancement factor of 0.0 gives a blurred image. A
+    factor of 1.0 gives the original image. And a factor
+    of 2.0 gives a sharpened image. It blends the source
+    image and the degenerated mean image:
+
+    ``output = img * factor + degenerated * (1 - factor)``
```
Comment (ZwwWayne) [inline mmcv/image/photometric.py:239]: May use code block or math block for this sentence.

File: mmcv/image/photometric.py [hunk 3]
```diff
@@ -227,6 +227,34 @@ def adjust_contrast(img, factor=1.):
...
+
+    Args:
+        img (ndarray): Image to be sharpened. BGR order.
+        factor (float): Same as :func:`mmcv.adjust_brightness`.
+
+    Returns:
+        ndarray: The sharpened image.
+    """
+    # adopted from PIL.ImageFilter.SMOOTH
+    kernel = np.array([[1., 1., 1.], [1., 5., 1.], [1., 1., 1.]]) / 13
```
Comment (ZwwWayne) [inline mmcv/image/photometric.py:249]: Should we make the kernel a settable  argument and by default using this value?

Comment (LXXXXR) [reply mmcv/image/photometric.py]: Revised accordingly. However, one thing to note is that there is no way to check if the kernel users set can actually perform the function of adjusting sharpness, so the function `adjust_sharpness` will not always performs the function its name indicates but performs whatever transform determined by the kernel.

Comment (ZwwWayne) [reply mmcv/image/photometric.py]: This can be put into the Notification section in the docstring as a reminder.

**Review by ZwwWayne — APPROVED** [review]
LGTM

**Pull request merged**